html/modules/custom/reliefweb_import/src/Plugin/ReliefWebImporter/EchoDataImporter.php

```diff
@@ -0,0 +1,391 @@
+
+    $code = strtolower($code);
+    return $disaster_mapping[$code] ?? NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state): array {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $form['api_url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('API URL'),
+      '#description' => $this->t('The base URL of the UNHCR Data API.'),
+      '#default_value' => $form_state->getValue('api_url', $this->getPluginSetting('api_url', '', FALSE)),
+      '#required' => TRUE,
+    ];
+
+    $form['timeout'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Timeout'),
+      '#description' => $this->t('Connection and request timeout in seconds.'),
+      '#default_value' => $form_state->getValue('timeout', $this->getPluginSetting('timeout', 10, FALSE)),
+      '#min' => 1,
+      '#required' => TRUE,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function importContent(int $limit = 50): bool {
+    // Get list of documents.
+    try {
+      $provider_uuid = $this->getPluginSetting('provider_uuid');
+
+      // Retrieve the POST API content processor plugin.
+      $plugin = $this->contentProcessorPluginManager->getPluginByResource('reports');
+
+      // Ensure the provider is valid.
+      $plugin->getProvider($provider_uuid);
+
+      $this->getLogger()->info('Retrieving documents from the UNHCR data API.');
+
+      // Retrieve the latest created documents.
+      $documents = $this->getDocuments($limit);
+
+      if (empty($documents)) {
+        $this->getLogger()->notice('No documents.');
+        return TRUE;
+      }
+    }
+    catch (\Exception $exception) {
+      $this->getLogger()->error($exception->getMessage());
+      return FALSE;
+    }
+
+    $this->getLogger()->info(strtr('Retrieved @count Echo Flash documents.', [
+      '@count' => count($documents),
+    ]));
+
+    // Sort the documents by ID ascending to process the oldest ones first.
+    ksort($documents);
+
+    // Process the documents importing new ones and updated ones.
+    $processed = $this->processDocuments($documents, $provider_uuid, $plugin);
+
+    // @todo check if we want to return TRUE only if there was no errors or if
+    // return TRUE for partial success is fine enough.
+    return $processed > 0;
+  }
+
+  /**
+   * Retrieve documents from the Echo Flash API.
+   *
+   * @param int $limit
+   *   Maximum number of documents to retrieve at once.
+   * @param string $order_property
+   *   Property to use to sort the documents.
+   *
+   * @return array
+   *   List of documents keyed by IDs.
+   */
+  protected function getDocuments(int $limit, string $order_property = 'created'): array {
+    // Get list of documents.
+    try {
+      $timeout = $this->getPluginSetting('timeout', 10, FALSE);
+      $api_url = $this->getPluginSetting('api_url');
+
+      $response = $this->httpClient->get($api_url, [
```

Comment:
This doesn't handle the `limit`. This could be passed via the `ItemsPageSize` parameter.

Reply:
It's part of the URL, tested with `https://erccportal.jrc.ec.europa.eu/API/ERCC/EchoFlash/GetPagedItems?ItemsCurrentPageIndex=1&ItemsPageSize=50&Filter%5BStatus%5D%5B%5D=Published`